lib/logstash/filters/dictionary/file.rb

```diff
@@ -0,0 +1,136 @@
+# encoding: utf-8
+require 'rufus-scheduler'
+require "logstash/util/loggable"
+require "logstash/filters/fetch_strategy/file"
+
+java_import 'java.util.concurrent.locks.ReentrantReadWriteLock'
+
+module LogStash module Filters module Dictionary
+  class File
+    def self.create(path, refresh_interval, refresh_behaviour, exact, regex)
+      if /\.y[a]?ml$/.match(path)
+        instance = YamlFile.new(path, refresh_interval, exact, regex)
+      elsif path.end_with?(".json")
+        instance = JsonFile.new(path, refresh_interval, exact, regex)
+      elsif path.end_with?(".csv")
+        instance = CsvFile.new(path, refresh_interval, exact, regex)
+      else
+        raise "Translate: Dictionary #{path} has a non valid format"
+      end
+      if refresh_behaviour == 'merge'
+        instance.set_update_strategy(:merge_dictionary)
+      elsif refresh_behaviour == 'replace'
+        instance.set_update_strategy(:replace_dictionary)
+      else
+        # we really should never get here
+        raise(LogStash::ConfigurationError, "Unknown value for refresh_behaviour=#{refresh_behaviour.to_s}")
+      end
+    end
+
+    include LogStash::Util::Loggable
+    attr_reader :dictionary, :fetch_strategy
+
+    def initialize(path, refresh_interval, exact, regex)
+      @dictionary_path = path
+      @refresh_interval = refresh_interval
+      @short_refresh = @refresh_interval < 300.001
+      rw_lock = java.util.concurrent.locks.ReentrantReadWriteLock.new
+      @write_lock = rw_lock.writeLock
+      @dictionary = Hash.new()
+      @update_method = method(:merge_dictionary)
+      initialize_for_file_type
+      raise_exception = true
+      if exact
+        if regex
+          @fetch_strategy = FetchStrategy::File::ExactRegex.new(@dictionary, rw_lock)
+        else
+          @fetch_strategy = FetchStrategy::File::Exact.new(@dictionary, rw_lock)
+        end
+      else
+        @fetch_strategy = FetchStrategy::File::RegexUnion.new(@dictionary, rw_lock)
+      end
+      load_dictionary(raise_exception)
```

Comment:
instead of declaring `raise_exception = true` above for the only purpose of passing it here by name, you could also remove the above statement and just do 
```
load_dictionary(raise_exception = true)
```
if the intent is to be explicit on the parameter purpose. otherwise
```
load_dictionary(true)
```

Reply:
This code was from before and I think the intention is to reveal the meaning behind the boolean.
I think I should add a comment.
```
load_dictionary(true) # true here means raise an exception once on initial load and not thereafter
```

Reply:
yup or a constant or as I suggested just `load_dictionary(raise_exception = true)` which is a poor's man named parameters. It is completely useless code-wise but serves the purpose of describing the intent of the boolean param.